tests/ha/test_control_pod_restart.py

```diff
@@ -1289,3 +1293,411 @@ def test_copy_obj_after_ctrl_pod_failover(self):
+                assert_utils.assert_true(False, "Please check background process logs")
+        LOGGER.info("Waiting for %s seconds to perform copy object ",
+                    HA_CFG["common_params"]["20sec_delay"])
+        time.sleep(HA_CFG["common_params"]["20sec_delay"])
+
+        LOGGER.info("Control pod %s is hosted on %s node", self.control_pod_name,
+                    self.control_node)
+
+        failover_node = self.system_random.choice([ele for ele in self.host_worker_list if ele !=
+                                                   self.control_node])
+        LOGGER.debug("Fail over node is: %s", failover_node)
+        event.set()
+        LOGGER.info("Step 3: Failover control pod %s to node %s and check cluster status",
+                    self.control_pod_name, failover_node)
+        pod_yaml = {self.control_pod_name: self.modified_yaml}
+        resp = self.ha_obj.failover_pod(pod_obj=self.node_master_list[0], pod_yaml=pod_yaml,
+                                        failover_node=failover_node)
+        assert_utils.assert_true(resp[0], resp)
+        LOGGER.info("Step 3: Successfully failed over control pod to %s. Cluster is in good state",
+                    failover_node)
+        event.clear()
+
+        self.restore_pod = self.deploy = True
+        LOGGER.info("Step 4: Verify if IAM users %s are persistent across control pod failover",
+                    uids)
+        for user in uids:
+            resp = self.rest_iam_user.get_iam_user(user)
+            assert_utils.assert_equal(int(resp.status_code), HTTPStatus.OK.value,
+                                      f"Couldn't find user {user} after control pod failover")
+            LOGGER.info("User %s is persistent: %s", user, resp)
+        LOGGER.info("Step 4: Verified all IAM users %s are persistent across control pod "
+                    "failover", uids)
+
+        LOGGER.info("Step 5: Checking responses from background process")
+        thread.join()
+        responses = tuple()
+        while len(responses) < 3:
+            responses = output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+
+        if not responses:
+            assert_utils.assert_true(False, "Background process failed to do copy object")
+
+        put_etag = responses[0]
+        exp_fail_bkt_obj_dict = responses[1]
+        failed_bkts = responses[2]
+        LOGGER.debug("Responses received from background process:\nput_etag: "
+                     "%s\nexp_fail_bkt_obj_dict: %s\nfailed_bkts: %s", put_etag,
+                     exp_fail_bkt_obj_dict, failed_bkts)
+        assert_utils.assert_true(len(failed_bkts) != 0, "Failed to do copy object when control "
+                                                        "pod restart. Failed "
+                                                        f"buckets: {failed_bkts}")
+        assert_utils.assert_true(len(exp_fail_bkt_obj_dict), f"Failed buckets "
```

Comment:
same as above

Reply:
In this also no expected bucket object failed, that's why assert.